en/concept/events_interface.md

```diff
@@ -0,0 +1,128 @@
+# Events Interface
+
+The *Events Interface* provides a system-wide API to inform a user about certain events.
+It is meant to replace existing usage of `mavlink_log_*` calls.
+
+## Usage
+### Basic
+To use the API, add this include:
+```cpp
+#include <px4_platform_common/events.h>
+```
+And then define and send the event from the desired code location:
+```cpp
+events::send(events::ID("mymodule_test"), events::Log::Info, "Test Message");
+```
+
+#### Backward compatibility
+For older GCS versions without events interface support, PX4 currently sends out all events also as `mavlink_log_*` `STATUSTEXT` message.
+In addition, the message must be tagged with an appended tab (`\t`) so that newer GCS's can ignore that and only show the event.
+
+So whenever adding an event, be sure to also add a `mavlink_log_` call. For example:
+```cpp
+mavlink_log_info(mavlink_log_pub, "Test Message\t");
+events::send(events::ID("mymodule_test"), events::Log::Info, "Test Message");
+```
+All such `mavlink_log_` calls will be removed after the next release.
+
+### Detailed
+The above is a minimal example, this is a more extensive one:
+```cpp
+uint8_t arg1 = 0;
+float arg2 = -1.f;
+/* EVENT
+ * @description
+ * This is the detailed event description.
+ *
+ * - value of arg1: {1}
+ * - value of arg2: {2:.1}
+ *
+ * <profile name="dev">
+ * (This paragraph is only meant to be shown to developers).
+ * This behavior can be configured with the parameter <param>COM_EXAMPLE</param>.
+ * </profile>
+ *
+ * Link to documentation: <a>https://docs.px4.io</a>
+ */
+events::send<uint8_t, float>(events::ID("event_name"),
+	{events::Log::Error, events::LogInternal::Info}, "Event Message", arg1, arg2);
+```
+
+Explanations and requirements:
+- `/* EVENT`: this is an indicator for the parser that there is additional metadata following as comment.
```

Comment:
"the parser" - what parser?

Reply:
Or to put it another way, it isn't clear what events are generated into. I know we do generate the ids and metadata, but perhaps also docs.

Reply:
Would `metadata parser` be clearer for you? The output could be different things, docs could also be generated in future.

Reply:
Not really. At this point in the doc you'd either link to information about the parser OR you might modify the text from

> this is an indicator for the parser that there is additional metadata following as comment.

To

> This tag indicates the start of a metadata definition section.

The problem here is that when I saw parser, I wanted to know what the parser does and can do/cannot do. We need to document things like:
- What the parser generates
- we know component info file - is the format anywhere that can be linked to
- what about translation information? 
- docs aren't generated yet - we should say that, unless we plan to generate docs for the events in near future.

Does that make sense?

Reply:
Ok, updated